aspnetcore/blazor/lifecycle.md

```diff
@@ -19,29 +19,29 @@ The Blazor framework includes synchronous and asynchronous lifecycle methods. Ov
 
 ### Component initialization methods
 
-<xref:Microsoft.AspNetCore.Components.ComponentBase.OnInitializedAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnInitialized*> execute code that initializes a component. These methods are only called one time when the component is first instantiated.
+<xref:Microsoft.AspNetCore.Components.ComponentBase.OnInitializedAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnInitialized*> are invoked when the component is initialized after having received its initial parameters from its parent component. Use `OnInitializedAsync` when the component performs an asynchronous operation and should refresh when the operation is completed. These methods are only called one time when the component is first instantiated.
 
-To perform an asynchronous operation, use `OnInitializedAsync` and the `await` keyword on the operation:
+For a synchronous operation, override `OnInitialized`:
 
 ```csharp
-protected override async Task OnInitializedAsync()
+protected override void OnInitialized()
 {
-    await ...
+    ...
 }
 ```
 
-> [!NOTE]
-> Asynchronous work during component initialization must occur during the `OnInitializedAsync` lifecycle event.
-
-For a synchronous operation, use `OnInitialized`:
+To perform an asynchronous operation, override `OnInitializedAsync` and use the `await` keyword on the operation:
 
 ```csharp
-protected override void OnInitialized()
+protected override async Task OnInitializedAsync()
 {
-    ...
+    await ...
 }
 ```
 
+> [!NOTE]
+> Asynchronous work during component initialization must occur during the `OnInitializedAsync` lifecycle event.
```

Comment:
I don't completely understand the purpose of this note. I'd say that it's a *common pattern* to do this, but it isn't mandatory.

For example, you could trigger async initialization work during `SetParametersAsync` or `OnParametersSetAsync` or in an extreme case, even in the component's constructor (though in that case you have to fire-and-forget the operation, since constructors aren't async).

In other words, I'd suggest dropping this note as it's a bit misleading. As long as we've been clear that this is a common and recommended pattern, there's no need to use the word `must` about it.

Reply:
I think the purpose was **_in relation to using the non-async `OnInitialized`_**, but that really is *very obvious*. I'll drop it.